lib/ddtrace/tracer.rb

```diff
@@ -51,8 +51,11 @@ def services
     #
     def shutdown!
```

Comment:
Shutdown is not a safe operation to be executed in parallel, so we add a critical zone to it.
This affects our some of our concurrency tests.

Reply:
Is this related to #1063? Does it resolve it?

Reply:
It's the same issue.
I think the solution proposed in https://github.com/DataDog/dd-trace-rb/issues/1063 is actually better, as the writer is supposed to be a trusted thread-safe object. The tracer shouldn't have to make the writer thread-safe explicitly.
I'll remove this change so we can implement that other solution instead.